src/uct/tcp/tcp_net.c

```diff
@@ -24,18 +26,48 @@
 #include <netdb.h>
 
 
-typedef ssize_t (*uct_tcp_io_func_t)(int fd, void *data, size_t size, int flags);
+#define UCT_TCP_DO_IO_NB(_fd, _data, _length, _io_func, _status) \
```

Comment:
why changed function call to macros? seems less maintainable, and also unrelated to moving socket functions to another file in UCS